shared/src/ledger/native_vp/ethereum_bridge/bridge_pool_vp.rs

```diff
@@ -147,33 +157,54 @@ where
                     gas_check: EscrowDelta {
                         payer_account: &transfer.gas_fee.payer,
                         escrow_account: &BRIDGE_POOL_ADDRESS,
-                        expected_debit: debit,
-                        expected_credit: transfer.gas_fee.amount,
+                        expected_debit: EscrowAmount::Nam(debit),
+                        expected_credit: EscrowAmount::Nam(
+                            transfer.gas_fee.amount,
+                        ),
                     },
                     token_check: EscrowDelta {
                         payer_account: &transfer.transfer.sender,
                         escrow_account: &Address::Internal(
                             InternalAddress::EthBridge,
                         ),
-                        expected_debit: debit,
-                        expected_credit: transfer.transfer.amount,
+                        expected_debit: EscrowAmount::Nam(debit),
+                        expected_credit: EscrowAmount::Nam(escrow_amount),
                     },
                 }
             } else {
+                let escrow_amount = if native {
+                    EscrowAmount::Nam(
+                        Amount::try_from(transfer.transfer.amount).map_err(
+                            |e| {
+                                eyre!(
+                                    "Could not process the amount of NAM to \
+                                     be escrowed into the bridge pool: {:?}",
```

Comment:
nit: assuming this implements `Error`, it's best to display the actual error string
```suggestion
                                     be escrowed into the bridge pool: {}",
```